### PR TITLE
Migrate `src/runtime/callbacks` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

File: src/runtime/callbacks.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    );
+    return !!this.callbacks_[CallbackId.PAYMENT_RESPONSE];
+  }
+
+  hasPaymentResponsePending(): boolean {
+    return !!this.resultBuffer_[CallbackId.PAYMENT_RESPONSE];
+  }
+
+  setOnFlowStarted(
+    callback: ({flow, data}: {flow: string; data: unknown}) => void
```
Comment (justinchou-google) [inline src/runtime/callbacks.ts:201]: Should we be using type `object` here over `unknown`?

File: src/runtime/callbacks.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  ): void {
+    this.setCallback_(CallbackId.FLOW_STARTED, callback as Callback);
+  }
+
+  /**
+   * @return Whether the callback has been found.
+   */
+  triggerFlowStarted(flow: string, data = {}): boolean {
```
Comment (justinchou-google) [inline src/runtime/callbacks.ts:209]: similarly here, not sure if type for `data` needs to be explicitly defined.

File: src/runtime/callbacks.ts (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/callbacks.ts:201]: Should I update `setOnFlowStarted` in `src/api/subscriptions.ts` to match?

File: src/runtime/callbacks.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/runtime/callbacks.ts:209]: In this case, the default value implies `{}`. What do you think?
image

Comment (justinchou-google) [reply src/runtime/callbacks.ts:209]:
```
Object has similarities to both {} and object, but is slightly looser. Use {} for a type that include everything except null and undefined
```

Seems good to me!

File: src/runtime/callbacks.ts (hunk 1 above)
Comment (justinchou-google) [reply src/runtime/callbacks.ts:201]: Yes please!

File: src/runtime/callbacks.ts [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    return this.trigger_(CallbackId.FLOW_STARTED, {
+      flow,
+      data,
+    });
+  }
+
+  setOnFlowCanceled(
+    callback: (params: {flow: string; data: unknown}) => void
```
Comment (justinchou-google) [inline src/runtime/callbacks.ts:217]: `data` can be updated here, too, and similarly for `subscriptions.ts`

Comment (ChrisAntaki) [reply src/runtime/callbacks.ts]: Updated!

File: src/runtime/callbacks.ts (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/callbacks.ts]: Updated!

**Review by justinchou-google — APPROVED** [review]

**Pull request merged**